src/dist/replication/lib/replica_stub.cpp

```diff
@@ -1231,7 +1257,36 @@ void replica_stub::close()
         _log->close();
         _log = nullptr;
     }
+}
 
+std::string replica_stub::get_replica_dir(const char* app_type, global_partition_id gpid) const
```

Comment:
But the current strategy requires to handle duplication and conflicts, which is non-trivial. I think extensibility can be achieved by moving the replica directories on replica-stub initialization if changes are made to the data directories. How do you think?